lib/src/client.dart

```diff
@@ -146,10 +146,16 @@ class OdrsClient {
 
   Future<T?> _response<T>(HttpClientResponse response) async {
     final data = await response.transform(utf8.decoder).join();
-    final json = jsonDecode(data);
-    if (json is Map && json['success'] == false) {
-      throw OdrsException(json['msg'] as String);
+    try {
+      final json = jsonDecode(data);
+      if (json is Map && json['success'] == false) {
+        throw OdrsException(json['msg'] as String);
```

Comment:
is null also a corret value?

Reply:
for `json['msg']`? i don't know but we can probably find the source code somewhere. :) did you receive such a response that had `"success": false` but no message since you're asking?

Reply:
I just thought if json['success'] == null then the exception would not be thrown but if null is a correct value then this is np I guess :) ?
I didn't test the code yet, just looked over the commit

Reply:
i'm not sure if i'm looking at the correct code but technically, the [error reporting helper function](https://gitlab.gnome.org/Infrastructure/odrs-web/-/blob/master/odrs/util.py) in odrs-web does seem to allow omitting the message.
```py
def json_error(msg=None, errcode=400):
    """Error handler: JSON output"""
    item = {}
    item["success"] = False
    if msg:
        item["msg"] = msg
    dat = json.dumps(item, sort_keys=True, indent=4, separators=(",", ": "))
    return Response(response=dat, status=errcode, mimetype="application/json")
```

however, the helper method is not called without a message anywhere in the codebase... :thinking: 
```
odrs/util.py:def json_error(msg=None, errcode=400):
odrs/views_api.py:    json_error,
odrs/views_api.py:        return json_error(str(e))
odrs/views_api.py:            return json_error("invalid data, expected %s" % key)
odrs/views_api.py:            return json_error("missing data, expected %s" % key)
odrs/views_api.py:        return json_error("the user_hash is invalid")
odrs/views_api.py:        return json_error("summary is too long")
odrs/views_api.py:        return json_error("description is too long")
odrs/views_api.py:            return json_error("%s is not a valid string" % key)
odrs/views_api.py:            return json_error("account has been disabled due to abuse")
odrs/views_api.py:        return json_error("already reviewed this app")
odrs/views_api.py:        return json_error("summary is too short")
odrs/views_api.py:        return json_error("description is too short")
odrs/views_api.py:        return json_error("review contains taboo word")
odrs/views_api.py:        return json_error(str(e))
odrs/views_api.py:            return json_error("invalid data, expected %s" % key)
odrs/views_api.py:            return json_error("missing data, expected %s" % key)
odrs/views_api.py:        return json_error("the user_hash is invalid")
odrs/views_api.py:        return json_error("no user for {}".format(user_hash))
odrs/views_api.py:        return json_error(str(e))
odrs/views_api.py:            return json_error("invalid data, required %s" % key)
odrs/views_api.py:            return json_error("missing data, expected %s" % key)
odrs/views_api.py:        return json_error("the user_hash is invalid")
odrs/views_api.py:        return json_error("the user_skey is invalid")
odrs/views_api.py:            return json_error("account has been disabled due to abuse")
odrs/views_api.py:            return json_error("all negative karma used up")
odrs/views_api.py:        return json_error("invalid user_skey")
odrs/views_api.py:        return json_error("already voted on this app")
odrs/views_api.py:        return json_error("invalid review ID")
odrs/views_api.py:        return json_error(str(e))
odrs/views_api.py:            return json_error("invalid data, required %s" % key)
odrs/views_api.py:            return json_error("missing data, expected %s" % key)
odrs/views_api.py:        return json_error("the user_hash is invalid")
odrs/views_api.py:        return json_error("the user_skey is invalid")
odrs/views_api.py:        return json_error("no review")
odrs/views_api.py:        return json_error("no review")
odrs/views_api.py:        return json_error("the app_id is invalid")
odrs/views_api.py:        return json_error("invalid user_skey")
```

Reply:
If you're not sure you could eventually also check the http response code to be 200.
Sorry if this is not helpful